polars/polars-core/src/chunked_array/strings/encoding.rs

```diff
@@ -3,26 +3,63 @@ use {base64, hex};
 use crate::prelude::*;
 
 impl Utf8Chunked {
-    #[cfg(feature = "string_encoding")]
+    #[cfg(not(feature = "binary_encoding"))]
+    pub fn hex_decode(&self, strict: Option<bool>) -> PolarsResult<Utf8Chunked> {
+        let ca = self.apply_on_opt(|e| {
+            e.and_then(|s| {
+                hex::decode(s)
+                    // Safety
+                    // We already know that it is a valid utf8.
+                    .map(|bytes| Some(unsafe { String::from_utf8_unchecked(bytes) }.into()))
+                    .unwrap_or(None)
+            })
+        });
+
+        if strict.unwrap_or(false) && (ca.null_count() != self.null_count()) {
+            Err(PolarsError::ComputeError("Unable to decode inputs".into()))
+        } else {
+            Ok(ca)
+        }
+    }
+
+    #[cfg(feature = "binary_encoding")]
     pub fn hex_decode(&self, strict: Option<bool>) -> PolarsResult<BinaryChunked> {
         self.cast_unchecked(&DataType::Binary)?
             .binary()?
             .hex_decode(strict)
     }
-    #[cfg(feature = "string_encoding")]
+
     #[must_use]
     pub fn hex_encode(&self) -> Utf8Chunked {
         self.apply(|s| hex::encode(s).into())
     }
 
-    #[cfg(feature = "string_encoding")]
+    #[cfg(not(feature = "binary_encoding"))]
+    pub fn base64_decode(&self, strict: Option<bool>) -> PolarsResult<Utf8Chunked> {
+        let ca = self.apply_on_opt(|e| {
+            e.and_then(|s| {
+                base64::decode(s)
+                    // Safety
+                    // We already know that it is a valid utf8.
+                    .map(|bytes| Some(unsafe { String::from_utf8_unchecked(bytes) }.into()))
```

Comment:
Is every valid `utf8` also a valid `utf8` when we hex/base64 decode? I am not convinced of this.

I think we should do `from_utf8` so that utf8 validation runs.

Other than that we also should not heap allocate a new `String` but use a `str` instead.

And finally I think instead of `map().unwrap(None)` we can do `and_then` or `ok().and_then`.

Reply:
fixed in last commit